www/base/src/app/buildrequests/pendingbuildrequests.controller.coffee

```diff
@@ -0,0 +1,16 @@
+class Pendingbuildrequests extends Controller
+    constructor: ($log, $scope, dataService, bbSettingsService) ->
+        $scope.settings = bbSettingsService.getSettingsGroup("BuildRequests")
+        $scope.$watch('settings', ->
+            bbSettingsService.save()
+        , true)
+        buildrequestFetchLimit = $scope.settings.buildrequestFetchLimit.value
+
+        data = dataService.open().closeOnDestroy($scope)
+        $scope.buildrequests = data.getBuildrequests(limit: buildrequestFetchLimit, order:'-submitted_at', claimed:false)
+        $scope.buildrequests.onNew = (buildrequest) ->
```

Comment:
I didn't realize that, but this is terribly inefficient. Some buildbot instances have thousands of buildrequest pending, so browsing to that page will largely overload the master.